RNTester/js/components/ExamplePage.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @format
+ * @flow
+ */
+
+'use strict';
+
+import React from 'react';
+import {
+  ScrollView,
+  StyleSheet,
+  View,
+  Text,
+  Dimensions,
+  Image,
+} from 'react-native';
+
+import HeadingText from './HeadingText';
+import Background from './Background';
+import UseCase from './UseCase';
+
+type Props = $ReadOnly<{|
+  children?: React.Node,
+  title?: ?string,
+  description?: ?string,
+  ios?: ?boolean,
+  android?: ?boolean,
+|}>;
+
+const ScreenHeight = Dimensions.get('window').height;
+const ScreenWidth = Dimensions.get('window').width;
+
+export default function ExamplePage(props: Props) {
+  const title = props.title ? <HeadingText text={props.title} /> : null;
+
+  const description = props.description ?? '';
+  const androidImage = props.android ? (
+    <Image
+      style={{height: 35, width: 30, margin: 2}}
+      source={imagePaths.android}
+    />
+  ) : null;
+
+  const appleImage = props.ios ? (
+    <Image style={{height: 35, width: 30, margin: 2}} source={imagePaths.ios} />
+  ) : null;
+
+  const docsImage = <Image source={imagePaths.docs} />;
+
+  const children = React.Children.map(props.children, child => (
```

Comment:
I feel this makes it a bit too implicit.  Wouldn't it be better to import the `UseCase` component in each screen itself? This would also provide a lot of flexibility in terms of customizing the `UseCase` component on a per-screen basis (If we need to).

Reply:
@ani4aniket , what do you mean by this?
> Wouldn't it be better to import the UseCase component in each screen itself?

Are you suggesting that we just render the children w/o mapping them in `UseCase` tag?



We can pass styling in props, but idk if it's a good practice or not.
> This would also provide a lot of flexibility in terms of customizing the UseCase component on a per-screen basis

Reply:
> I feel this makes it a bit too implicit. Wouldn't it be better to import the `UseCase` component in each screen itself? This would also provide a lot of flexibility in terms of customizing the `UseCase` component on a per-screen basis (If we need to).

@anku255 As this is a placeholder/Card for each UseCase of a RN Component, I can't think of a scenario where there will be any need to customize it and it be easier if all screens follow the same pattern. But calling it on every screen would work if you any other scenario in mind..

Reply:
> I feel this makes it a bit too implicit. Wouldn't it be better to import the `UseCase` component in each screen itself? This would also provide a lot of flexibility in terms of customizing the `UseCase` component on a per-screen basis (If we need to).

@anku255 As this is a placeholder/Card for each UseCase of a RN Component, I can't think of a scenario where there will be any need to customize it and it'll be easier if all screens follow the same pattern. But calling it on every screen would work if you have any other scenario in mind..

Reply:
Yes, that's what I meant. If you look at @jevakallio example

```js
<ExamplePage title="Alert" description="Launches an alert dialog with the specified title and message" android ios>
  <UseCase title="Alert with 2 buttons" android ios>
    // use case content here
  </UseCase>
  <UseCase title="Alert with 3 buttons" ios>
    // use case content here
  </UseCase>
   // etc
</ExamplePage>
```

The `UseCase` component is being imported in the Screen itself and not inside `ExamplePage`. I feel this makes it easier to see the different usecases in the screen itself.

Also, I don't know if we need to customize `UseCase` on per-screen basis but this pattern will allow us to do that if we ever needed to.

Reply:
@anku255 I guess mapping does the same thing and it will be better practice to follow that.

Reply:
@anku255 , I do understand the concern that we will face issues if we have to add custom styling to each `UseCase`, however, the `UseCase` component is just for the spacing and borders(if any) and will have uniform styling throughout.

It's just a simple map polyfil, we will remove it incase the usecase changes?

Reply:
@manyaagarwal While mapping does accomplish the same thing but there is a lot going on underneath the hood which you cannot tell from the Screen Page.

Let me explain what I mean. 

In `AlertExample.js`

You would write the usecases as follows (as per the current code): 

```js
<ExamplePage>
 {/* Use case 1 */}
  <View android ios note="note" title="title">

  </View>
{/* Use case 2 */}
  <View android ios note="note" title="title">

  </View>

</ExamplePage>

```


If you didn't know the implementation of `ExamplePage.js` you wouldn't even know how many `usecases` are in `AlertExample.js` without the comments. Also, if you see we are passing invalid props to `View`which doesn't really make sense unless you know they will be passed down to `UseCase` component inside `ExamplePage`.


On the other hand, if we imported `UseCase` in the screen itself, it the props would make more sense and also it would be easier to differentiate between usecases.

Reply:
@anku255 I am a little confused, why will we pass those to `View` ?  Isn't the code using a list of UseCases passed to the `ExamplePage` and is then mapping them to `UseCase` components.

Reply:
@anku255 are you saying that we should have different files for each use case

Reply:
@ani4aniket , this makes sense. I'll update the PR.

> if we imported UseCase in the screen itself, it the props would make more sense and also it would be easier to differentiate between usecases.